qiskit_ibm/managed/managedjob.py

```diff
@@ -120,9 +118,9 @@ def _async_submit(
                         job_tags=job_tags,
                         **run_config)
                 except IBMQBackendJobLimitError:
-                    final_states = [state.value for state in API_JOB_FINAL_STATES]
-                    oldest_running = backend.jobs(limit=1, descending=False,
-                                                  db_filter={"status": {"nin": final_states}})
+                    all_job_statuses = [status.name for status in JobStatus]
+                    oldest_running = backend.jobs(
+                        limit=1, descending=False, status=all_job_statuses)
```

Comment:
Sorry if it wasn't clear earlier, we still want to filter by non-final statuses here.

i.e., all job statuses - final statuses = non-final statuses

final statuses tuple is here, subtract this from all job statuses to get non-final statuses
https://github.com/Qiskit/qiskit-terra/blob/b757914b7200b2a4a4e8065477342f25a544df83/qiskit/providers/jobstatus.py#L30